plugins/scaffolder/src/components/TemplatePage/TemplatePage.tsx

```diff
@@ -93,11 +93,10 @@ export const createValidator = (
         const fieldName =
           isObject(propSchema) && (propSchema['ui:field'] as string);
         if (fieldName && typeof validators[fieldName] === 'function') {
-          validators[fieldName]!(
-            propData as JsonValue,
-            propValidation,
-            context,
-          );
+          validators[fieldName]!(propData as JsonValue, propValidation, {
+            ...context,
+            uiOptions: propSchema['ui:options'] ?? {},
```

Comment:
Dunno if it matters much, but would it make more sense to default to `undefined`? Thinking that otherwise we may want to make `uiOptions` required in the type